File: problems/3456-find-special-substring-of-length-k/solution.java
class Solution {
    public boolean hasSpecialSubstring(String s, int k) {
        for (int i = 0; i <= s.length() - k; i++) {
            char c = s.charAt(i);
            boolean same = true;
            for (int j = i + 1; j < i + k; j++) {
                if (s.charAt(j) != c) {
                    same = false;
                    break;
                }
            }
            if (same) {
                boolean beforeDifferent = true;
                if (i > 0 && s.charAt(i - 1) == c) {
                    beforeDifferent = false;
                }
                boolean afterDifferent = true;
                if (i + k < s.length() && s.charAt(i + k) == c) {
                    afterDifferent = false;
                }
                if (beforeDifferent && afterDifferent) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: problems/3456-find-special-substring-of-length-k/solution.c
bool hasSpecialSubstring(char* s, int k) {
    int n = 0;
    while (s[n] != '\0') {
        n++;
    }

    if (k > n) {
        return false;
    }

    for (int i = 0; i <= n - k; i++) {
        char firstChar = s[i];
        bool allSame = true;
        for (int j = 1; j < k; j++) {
            if (s[i + j] != firstChar) {
                allSame = false;
                break;
            }
        }

        if (allSame) {
            bool beforeDifferent = true;
            if (i > 0) {
                if (s[i - 1] == firstChar) {
                    beforeDifferent = false;
                }
            }

            bool afterDifferent = true;
            if (i + k < n) {
                if (s[i + k] == firstChar) {
                    afterDifferent = false;
                }
            }

            if (beforeDifferent && afterDifferent) {
                return true;
            }
        }
    }

    return false;
}